images/metering-ansible-operator/roles/meteringconfig/defaults/main.yml

```diff
@@ -461,7 +515,7 @@ _tls_overrides:
 
 
 # check if the user's spec contains the top-level tls.enabled field. if true, use the value stored in that field, else default to the value stored in values.yaml
-meteringconfig_tls_enabled: "{{ meteringconfig_default_values | combine(meteringconfig_spec_overrides, recursive=True) | json_query('tls.enabled') }}"
+meteringconfig_tls_enabled: "{{ meteringconfig_default_values | combine(meteringconfig_spec_overrides, recursive=True) | json_query('tls.enabled') and not meteringconfig_ocp_disabled }}"
```

Comment:
Hmm, I dislike this. I feel like we may want to reorganize the `meteringconfig_spec` overrides ordering so this isn't necessary and so that disable OCP takes precedence over it. Thinking...

Reply:
I can do some more manual testing to cover this case, and in the case you want to go away from this implementation, I would need to update the `_ocp_disabled_overrides` to include all the sections that `_tls_overrides` handles too.

Reply:
I don't understand that, why can't we just have 
```
meteringconfig_tls_enabled: "{{ meteringconfig_default_values | combine(_meteringconfig_ocp_disabled_overrides, meteringconfig_spec_overrides, recursive=True) | json_query('tls.enabled') }}"
```

`_meteringconfig_ocp_disabled_overrides` will have `tls.enabled` false, thus `meteringconfig_tls_enabled` will be false, thus, `_tls_overrides` will be `{}`.

Reply:
Yeah that would definitely do the trick - I was under the impression you didn't want to link the `meteringconfig_tls_enabled` and the value of the `meteringconfig_ocp_disabled`, and instead, only modify the order that their respective dictionaries appear in `meteringconfig_spec`.

Reply:
That's correct, I wanted to decouple them, but I think we have no choice, but this is a little more general than the previous approach. They're intrinsically related, so we must know the values of the other. But at least this makes the way we do it consistent with how we check values elsewhere.